src/contracts/core/AllocationManager.sol

```diff
@@ -256,82 +375,133 @@ contract AllocationManager is
-        uint64 _encumberedMagnitude = encumberedMagnitude[operator][strategy];
-
-        // If the pending change can't be cleared yet
-        if (block.timestamp < mInfo.effectTimestamp) {
-            return PendingMagnitudeInfo({
-                encumberedMagnitude: _encumberedMagnitude,
-                currentMagnitude: mInfo.currentMagnitude,
-                pendingDiff: mInfo.pendingDiff,
-                effectTimestamp: mInfo.effectTimestamp
-            });
+        bytes32 operatorSetKey,
+        IStrategy strategy
+    ) internal view returns (StrategyInfo memory, Allocation memory) {
+        StrategyInfo memory info = StrategyInfo({
+            maxMagnitude: _maxMagnitudeHistory[operator][strategy].latest(),
+            encumberedMagnitude: encumberedMagnitude[operator][strategy]
+        });
+
+        Allocation memory allocation = allocations[operator][operatorSetKey][strategy];
+
+        // If the pending change can't be completed yet, return as-is
+        if (block.number < allocation.effectBlock) {
+            return (info, allocation);
         }
 
-        // Pending change can be cleared - add delta to current magnitude
-        info.currentMagnitude = _addInt128(mInfo.currentMagnitude, mInfo.pendingDiff);
-        info.encumberedMagnitude = _encumberedMagnitude;
-        info.effectTimestamp = 0;
-        info.pendingDiff = 0;
+        // Otherwise, complete the pending change and return updated info
+        allocation.currentMagnitude = _addInt128(allocation.currentMagnitude, allocation.pendingDiff);
 
-        // If the cleared change was a deallocation, update encumbered magnitude
-        if (mInfo.pendingDiff < 0) {
-            info.encumberedMagnitude = _addInt128(_encumberedMagnitude, mInfo.pendingDiff);
+        // If the completed change was a deallocation, update used magnitude
+        if (allocation.pendingDiff < 0) {
+            info.encumberedMagnitude = _addInt128(info.encumberedMagnitude, allocation.pendingDiff);
         }
 
-        return info;
+        allocation.effectBlock = 0;
+        allocation.pendingDiff = 0;
+
+        return (info, allocation);
     }
 
-    /// @notice Update the operator's magnitude info in storage and their encumbered magnitude.
-    function _updateMagnitudeInfo(
+    function _updateAllocationInfo(
         address operator,
-        IStrategy strategy,
         bytes32 operatorSetKey,
-        PendingMagnitudeInfo memory info
+        IStrategy strategy,
+        StrategyInfo memory info,
+        Allocation memory allocation
     ) internal {
-        _operatorMagnitudeInfo[operator][strategy][operatorSetKey] = MagnitudeInfo({
-            currentMagnitude: info.currentMagnitude,
-            pendingDiff: info.pendingDiff,
-            effectTimestamp: info.effectTimestamp
-        });
-
+        // Update encumbered magnitude
         encumberedMagnitude[operator][strategy] = info.encumberedMagnitude;
```

Comment:
Note: multiple `_updateAllocationInfo` calls redundantly overwriting encumberedMagnitude in clearDeallocationQueue. Something to look at if `modifyAllocations` or `clearDeallocationQueue` are too expensive to call

Reply:
Yes, this is true. It was pretty hard to avoid the duplicate writes while keeping the code clean, unfortunately.

I felt it was probably okay given that SSTORE is significantly cheaper in these cases (only 100 gas for followup writes).

But yes, we can look at changing this if needed.